### PR TITLE
decompile unk_0206015C

### PR DESCRIPTION
**Pull request by fyr3n**
decomp of unk_0206015C, since it is my first contribution, let me know if I can improve things for the future, thanks.

### CODE REVIEW

File: include/unk_0206015C.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#include "main.h"
+#include "unk_0204639C.h"
+#include "heap.h"
+#include "event_data.h"
+
+/*  Note to future reader, there might be some errors
+    in the return type / parameters. They are right for this code
+    but they might not be for every code, please do not blindly
+    trust these prototypes if you are decompiling these functions. */
+extern THUMB_FUNC u32   MOD06_02244DB0(u32);
```
Comment (red031000) [inline include/unk_0206015C.h:12]: externs in the .c file

File: include/unk_0206015C.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#include "global.h"
```
Comment (red031000) [inline include/unk_0206015C.h:1]: it's probably somewhat better to only put the includes required by the .h in the .h, and the rest of the includes in the .c

File: arm9/src/unk_0206015C.c [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        MOD06_02245190(unkStruct1->unk4);
+        MOD06_02244DBC(unkStruct1->unk0);
+        FreeToHeap(unkStruct1);
+        return 1;
+
+    default:
+        break;
+    }
+    return 0;
+}
```
Comment (red031000) [inline arm9/src/unk_0206015C.c:113]: newline at end

File: arm9/src/unk_0206015C.c [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            unkStruct1->action = 1;
+        }
+        break;
+
+    case 8:
+        FUN_0206367C(savStruct, 0);
+        MOD06_02245190(unkStruct1->unk4);
+        MOD06_02244DBC(unkStruct1->unk0);
+        FreeToHeap(unkStruct1);
+        return 1;
```
Comment (red031000) [inline arm9/src/unk_0206015C.c:107]: return TRUE

File: arm9/src/unk_0206015C.c [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+    default:
+        break;
+    }
+    return 0;
```
Comment (red031000) [inline arm9/src/unk_0206015C.c:112]: return FALSE

File: include/unk_0206015C.h [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    u32   unk4;
+    u32   unk8;
+    u8    action;
+    u8    unkD; // some kind of counter?
+    u8    unkE; // some kind of counter?
+};
+
+THUMB_FUNC void FUN_0206015C(struct UnkSavStruct80* savStruct);
+THUMB_FUNC BOOL FUN_02060194(struct UnkStruct_0204639C * unkStruct0);
+
```
Comment (red031000) [inline include/unk_0206015C.h:60]: header guards?

**Review by red031000 — CHANGES_REQUESTED** [review]
few things to change

**Comment by fyr3n** [comment]
changes done

**Comment by fyr3n** [comment]
changed commit name since it wasn't explicit enough.

**Review by red031000 — APPROVED** [review]

**Review by ProjectRevoTPP — APPROVED** [review]

**Pull request merged**